odl/solvers/functional/default_functionals.py

```diff
@@ -0,0 +1,452 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Default functionals defined on any space similar to R^n or L^2."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+
+from odl.solvers.functional.functional import Functional
+from odl.operator.operator import Operator
+from odl.solvers.advanced.proximal_operators import (
+    proximal_l1, proximal_cconj_l1, proximal_l2, proximal_cconj_l2,
+    proximal_l2_squared, proximal_zero)
+
+from odl.operator.default_ops import (ZeroOperator, ScalingOperator)
+
+
+__all__ = ('L1Norm', 'L2Norm', 'L2NormSquared', 'ZeroFunctional',
+           'ConstantFunctional')
+
+
+class L1Norm(Functional):
+
+    """The functional corresponding to L1-norm.
+
+    The L1-norm, ``||x||_1``,  is defined as the integral/sum of ``|x|``.
+
+    Notes
+    -----
+    If the functional is defined on an :math:`\mathbb{R}^n`-like space, the
+    :math:`L_1`-norm is defined as
+
+    .. math::
+
+        \| x \|_1 = \\sum_{i = 1}^n |x_i|.
+
+    If the functional is defined on an :math:`L_2`-like space, the
+    :math:`L_1`-norm is defined as
+
+    .. math::
+
+        \| x \|_1 = \\int_{\Omega} |x| dx.
+
+    """
+
+    def __init__(self, space):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` or `FnBase`
```

Comment:
I know. But after a short discussion with @kohr-h I chose to mention both. In order to hopefully make it less confusing for someone how wants to try it out for the first time (I don't think I have ever used `FnBase` explicitly myself so far...).

Keep both or only `FnBase`?